stan/math/rev/fun/singular_values.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef MATH_SINGULAR_VALUES_H
+#define MATH_SINGULAR_VALUES_H
+
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/functor/reverse_pass_callback.hpp>
+#include <stan/math/rev/core/arena_matrix.hpp>
+#include <stan/math/rev/fun/value_of.hpp>
+#include <stan/math/rev/fun/typedefs.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/err/check_symmetric.hpp>
+#include <stan/math/prim/err/check_nonzero_size.hpp>
+#include <stan/math/prim/fun/typedefs.hpp>
+#include <stan/math/prim/fun/value_of_rec.hpp>
+#include <Eigen/SVD>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return the singular values of the specified symmetric matrix.
+ *
+ * For MxN input matrix, M must be <= N
+ *
+ * @tparam EigMat type of input matrix
+ * @param m MxN input matrix, M <= N
+ * @return Singular values of matrix
+ */
+template <typename EigMat,
+	  require_eigen_vt<EigMat>* = nullptr,
+	  require_not_vt_var<EigMat>* = nullptr>
+inline auto singular_values(const EigMat& m) {
+  using ret_type = promote_scalar_t<var, Eigen::VectorXd>;
+  check_nonzero_size("singular_values", "m", m);
+
+  const int M = m.rows();
+  auto arena_m = to_arena(m);
+  
+  Eigen::MatrixXd m_val = value_of(arena_m);
+  Eigen::JacobiSVD<EigMat> svd(m_val, Eigen::ComputeFullU | Eigen::ComputeFullV);
```

Comment:
`Eigen::JacobiSVD<EigMat>` you're computing the SVD of an `Eigen::MatrixXd` so this should be `Eigen::JacobiSVD<Eigen::MatrixXd>`

`EigMat` could be `Eigen::Matrix<var, -1, -1>`, which is an autodiff type.